### PR TITLE
Add debug getbadblocks

### PR DESCRIPTION
**Pull request by cpurta**
Blockchain implementation will now have a blacklist of hashes that will allow for it to add bad blocks to a HashMap and in order to get all bad blocks we return a vector of hashes. This then allows for the rpc api to then return all of the current bad blocks that are within the current blockchain. 

This will resolve https://github.com/paritytech/parity-ethereum/issues/8761 when merged.

### CODE REVIEW

**Comment by cpurta** [comment]
[clabot:check]

**Comment by cpurta** [comment]
This currently does not build as of now. Issue is stemming from `rpc/src/v1/impls/eth.rs` L821. Currently working on getting that to work. Would appreciate a review of the merge request as a whole. 

I am still pretty new to Rust so there are probably some lines that can be refactored. Also if the naming of the new functions seems off I can rename those.

**Comment by cpurta** [comment]
Any chance that I can get a quick review on the changes here and some feedback on how to finish this?

**Comment by Tbaut** [comment]
Let me blindly ping @andresilva @debris @tomusdrw to attract their attention :)

File: ethcore/src/blockchain/blockchain.rs [hunk 1]
```diff
@@ -1364,6 +1371,26 @@ impl BlockChain {
...
+		if self.blacklisted_hashes().contains_key(&hash) {
+			return true
+		}
+		false
+	}
+
+	/// Get all blacklisted hashes
+	fn blacklisted_hashes(&self) -> HashMap<H256, bool> {
+		let mut blacklist = HashMap::new();
+		blacklist.insert("05bef30ef572270f654746da22639a7a0c97dd97a7050b9e252391996aaeb689".into(), true);
```
[the remark this answers is not in the document]
Comment (cpurta) [reply ethcore/src/blockchain/blockchain.rs]: These represent the list of bad hashes (usually hard forks) that determine whether a block is considered "bad". Similar to the implementation in geth: https://github.com/ethereum/go-ethereum/blob/master/core/blocks.go

File: rpc/src/v1/impls/eth.rs [hunk 2]
```diff
@@ -817,6 +817,16 @@ impl<C, SN: ?Sized, S: ?Sized, M, EM, T: StateInfo + 'static> Eth for EthClient<
 		Ok(true)
 	}
 
+	fn bad_blocks(&self, include_txs: bool) -> BoxFuture<Vec<RichBlock>> {
+		let blocks = self.client.bad_blocks()
+			.unwrap()
```
[the remark this answers is not in the document]
Comment (cpurta) [reply rpc/src/v1/impls/eth.rs]: Ahh. I didn't see that guide. I'll get on fixing that.

**Comment by cpurta** [comment]
@folsen addressed the unwrap issue and commented back on a couple of your comments. Would appreciate some feedback on comments and changes when you have some time.

**Comment by 5chdn** [comment]
Superseeded by #9433

**Pull request closed without merging**